### PR TITLE
fixes #852

### LINKED ISSUE #852: Translation of Filter of Date Columns
I noticed the Date Filter is shown only in English

image

Paul-Grist said the following on the topic in the forum

**Hmm someone needs to go into this code:**
https://github.com/gristlabs/grist-core/blob/d008a32eb3eef6fb7d233b8a0d85fced3209abcd/app/client/ui/DateRangeOptions.ts#L9-L41

**and mark the text there in the special way that makes it translatable. It is a small and hopefully easy piece of work..**

### PR DESCRIPTION
**Pull request by thomasweaver627**
#### Reference Issues
Translation of Filter of Date Columns #852

#### Changes
Added _makeT()_ function from localization.ts and declared _const t_

### CODE REVIEW

**Comment by thomasweaver627** [comment]
Fixed initial commit code where I did not wrap strings with t

**Review by fflorent — CHANGES_REQUESTED** [review]
This works perfectly, thanks! :pray:

One tiny thing: in order to send the translation keys to Weblate, you may run this command to refresh the localization strings in the dedicated json files at the root of the project repository:
```bash
yarn generate:translation
```

And then index and commit the `/static/locales/en.client.json` file.

NB: I am not an official reviewer. You should wait for a Grist Labs employee to review your code before having this PR merged.

File: app/client/ui/DateRangeOptions.ts
Comment (georgegevoian) [inline app/client/ui/DateRangeOptions.ts:12]: Could you change this to a function, `getDateRangeOptions`, that returns this object?

The reason is that the i18n library needs to be initialized before the function returned by `makeT` will work, so calling `t` at top-level is a bit risky.

**Review by georgegevoian — CHANGES_REQUESTED** [review]
Thanks for the PR @thomasweaver627! Just have one comment.

File: app/client/ui/DateRangeOptions.ts
Comment (georgegevoian) [reply app/client/ui/DateRangeOptions.ts:12]: The `makeT` part looks ok. The change I'm talking about is this one:

```javascript
export function getDateRangeOptions(): IDateRangeOption[] {
  return [
    {
      label: t('Today'),
      min: CURRENT_DATE,
      max: CURRENT_DATE,
    },
    {
      label: t('Last 7 days'),
      min: [{quantity: -7, unit: 'day'}],
      max: [{quantity: -1, unit: 'day'}],
    },
    {
      label: t('Next 7 days'),
      min: [{quantity: 1, unit: 'day'}],
      max: [{quantity: 7, unit: 'day'}],
    },
    {
      label: t('Last Week'),
      min: [{quantity: -1, unit: 'week'}],
      max: [{quantity: -1, unit: 'week', endOf: true}],
    },
    {
      label: t('Last 30 days'),
      min: [{quantity: -30, unit: 'day'}],
      max: [{quantity: -1, unit: 'day'}],
    },
    {
      label: t('This week'),
      min: [{quantity: 0, unit: 'week'}],
      max: [{quantity: 0, unit: 'week', endOf: true}],
    },
    {
      label: t('This month'),
      min: [{quantity: 0, unit: 'month'}],
      max: [{quantity: 0, unit: 'month', endOf: true}],
    },
    {
      label: t('This year'),
      min: [{quantity: 0, unit: 'year'}],
      max: [{quantity: 0, unit: 'year', endOf: true}],
    },
  ];
}
```

You'll just need to update any code that previously referenced `DateRangeOptions` to `getDateRangeOptions()`.

Comment (thomasweaver627) [reply app/client/ui/DateRangeOptions.ts:12]: Oh I see, that makes more sense. Thank you for clarifying!!

**Review by georgegevoian — APPROVED** [review]
Thanks @thomasweaver627!

**Comment by fflorent** [comment]
Ah right, I have just learnt that the update of the `/static/locales/en.client.json` file is handled automatically:
https://github.com/gristlabs/grist-core/pull/888

Congrats @thomasweaver627 for your first PR!

**Comment by thomasweaver627** [comment]
> Congrats @thomasweaver627 for your first PR!

Thank you @fflorent!!

**Pull request merged**
